rp2040-hal/src/rosc.rs

```diff
@@ -1,5 +1,16 @@
 //! Ring Oscillator (ROSC)
-// See [Chapter 2 Section 17](https://datasheets.raspberrypi.org/rp2040/rp2040_datasheet.pdf) for more details
+//! See [Chapter 2 Section 17](https://datasheets.raspberrypi.org/rp2040/rp2040_datasheet.pdf) for more details
+//!
+//! In addition to its obvious role as a clock source, [`RingOscillator`] can also be used as a random number source
+//! for the [`rand`] crate:
+//!
+//! ```no_run
+//! # let mut pac = rp2040_pac::Peripherals::take().unwrap();
+//! use rp2040_hal::rosc::RingOscillator;
+//! use rand::Rng;
+//! let mut rnd = RingOscillator::new(pac.ROSC).initialize();
+//! let random_value: u32 = rnd.gen();
+//! ```
 
```

Comment:
Add link to rand so the link on line 5 goes somewhere
```suggestion
//! ```
//! [`rand`]: https://docs.rs/rand
```